app/src/main/java/com/quarkworks/apartmentgroceries/service/SyncGroceryItem.java

```diff
@@ -124,4 +133,124 @@ public Void then(Task<JSONObject> task) throws Exception {
 
         return networkRequest.runNetworkRequest().continueWith(continuation);
     }
+
+    public static Task<JSONObject> addGroceryPhoto(final String groceryId, byte[] data) {
+
+
+        final Task<JSONObject>.TaskCompletionSource taskCompletionSource = Task.create();
+
+        Continuation<JSONObject, Void> addPhotoNameToGroceryPhoto = new Continuation<JSONObject, Void>() {
+            @Override
+            public Void then(Task<JSONObject> task) throws Exception {
+                if (task.isFaulted()) {
+                    Exception exception = task.getError();
+                    Log.e(TAG, "Error in uploadPhoto", exception);
+                    throw exception;
+                }
+
+                String photoName = task.getResult().optString("name");
+                if (TextUtils.isEmpty(photoName)) {
+                    throw new InvalidResponseException("Empty response");
+                }
+
+                final UrlTemplate template = UrlTemplateCreator.addGroceryPhoto(groceryId, photoName);
+
+                Continuation<JSONObject, Void> addingGroceryPhoto = new Continuation<JSONObject, Void>() {
+                    @Override
+                    public Void then(Task<JSONObject> task) throws Exception{
+                        if (task.isFaulted()) {
+                            Exception exception = task.getError();
+                            Log.e(TAG, "Error in addingGroceryPhoto", exception);
+                            throw exception;
```

Comment:
When this is successful, you need add a new RGroceryPhoto to Realm. After you create a new grocery item, the photos don't show up when you look at it in the ViewPager because there are no grocery photos in the database for that grocery item yet.